### PR TITLE
update semtech-udp to 0.8

### PR DESCRIPTION
**Pull request by lthiery**
`semtech-udp v0.8` has improvements to errors and uses the `macaddr` crate instead of the self-spun type.

### CODE REVIEW

File: src/gateway.rs [hunk 1]
```diff
@@ -80,8 +80,9 @@ impl Gateway {
 
     async fn handle_udp_event(&mut self, logger: &Logger, event: Event) -> Result {
         match event {
-            Event::UnableToParseUdpFrame(buf) => {
-                info!(logger, "ignoring semtech udp parsing error for {:?}", buf)
+            Event::UnableToParseUdpFrame(e, buf) => {
+                info!(logger, "ignoring semtech udp parsing error {e}");
+                info!(logger, "ignoring semtech udp parsing error for {buf:?}");
```
Comment (JayKickliter) [inline src/gateway.rs:85]: is there a reason to not combine this with the line above?

Comment (lthiery) [reply src/gateway.rs:85]: No that's a good point. Fixing now

File: src/gateway.rs [hunk 2]
```diff
@@ -80,8 +80,11 @@ impl Gateway {
 
     async fn handle_udp_event(&mut self, logger: &Logger, event: Event) -> Result {
         match event {
-            Event::UnableToParseUdpFrame(buf) => {
-                info!(logger, "ignoring semtech udp parsing error for {:?}", buf)
+            Event::UnableToParseUdpFrame(e, buf) => {
+                info!(
```
[the remark this answers is not in the document]
Comment (madninja) [reply src/gateway.rs:84]: ah yeah, that makes sense too.. if it's rare and really not expected this should be an error indeed

[the remark this answers is not in the document]
Comment (lthiery) [reply src/gateway.rs]: yeah, I think warning is right. fixed it

**Review by madninja — APPROVED** [review]

**Review by madninja — APPROVED** [review]

**Pull request merged**